premarum-backend/PreEnrollmentMgmt.Core/Services/StudentService.cs

```diff
@@ -34,4 +35,17 @@ public async Task UpdateDepartment(string studentEmail, int newDepartmentId)
         student.DepartmentId = newDepartmentId;
         _studentRepository.Save(student);
     }
+
+    public async Task<ICollection<CoursesTaken>> AddCoursesTaken(string studentEmail, int[] courseIds)
+    {
+        var student = await _studentValidationService.ValidateStudentExists(studentEmail);
```

Comment:
Add a flag for the ValidateStudentFunction to return the Student's coursesTaken. 

Refer to validatePreEnrollmentExists function.